packages/control-property-editor/src/panels/outline/Tree.tsx

```diff
@@ -0,0 +1,393 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import type { ReactElement } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import type { IGroup, IGroupRenderProps, IGroupHeaderProps } from '@fluentui/react';
+import { Icon } from '@fluentui/react';
+import { UIList } from '@sap-ux/ui-components';
+
+import { selectControl, reportTelemetry } from '@sap-ux-private/control-property-editor-common';
+import type { Control, OutlineNode } from '@sap-ux-private/control-property-editor-common';
+
+import type { RootState } from '../../store';
+import { IconName } from '../../icons';
+import type { ControlChanges, FilterOptions } from '../../slice';
+import { FilterName } from '../../slice';
+import { NoControlFound } from './NoControlFound';
+import { adaptExpandCollapsed, getFilteredModel, isSame } from './utils';
+import { ChangeIndicator } from '../../components/ChangeIndicator';
+
+interface OutlineNodeItem extends OutlineNode {
+    level: number;
+    path: string[];
+}
+
+export const Tree = (): ReactElement => {
+    const dispatch = useDispatch();
+    const [selection, setSelection] = useState<{ group: undefined | IGroup; cell: undefined | OutlineNodeItem }>({
+        group: undefined,
+        cell: undefined
+    });
+    const [collapsed, setCollapsed] = useState<IGroup[]>([]);
+    const filterQuery = useSelector<RootState, FilterOptions[]>((state) => state.filterQuery);
+    const selectedControl = useSelector<RootState, Control | undefined>((state) => state.selectedControl);
+    const controlChanges = useSelector<RootState, ControlChanges>((state) => state.changes.controls);
+    const model: OutlineNode[] = useSelector<RootState, OutlineNode[]>((state) => state.outline);
+    const { groups, items } = useMemo(() => {
+        const items: OutlineNodeItem[] = [];
+        const filteredModel = getFilteredModel(model, filterQuery);
+        return { groups: getGroups(filteredModel, items), items };
+    }, [model, filterQuery, selection]);
+    const theme =
+        document.getElementsByTagName('HTML')[0].getAttribute('data-theme') === 'high contrast'
```

Comment:
minor comment, if there is method `setThemeOnDocument` -> https://github.com/SAP/open-ux-tools/blob/feat/1149/publishControlPropertyEditor/packages/control-property-editor/src/components/ThemeSelectorCallout.tsx#L123 then maybe makes sense to create similar on same level for `getThemeOnDocument`?

Reply:
Changed to retrieve theme from local storage instead [3d6b592](https://github.com/SAP/open-ux-tools/pull/1197/commits/3d6b5925479e8e9232c6e76d611038d40a3aeaea).